zulipterminal/ui_tools/buttons.py

```diff
@@ -288,11 +288,36 @@ def handle_link(self, *_: Any) -> None:
         """
         if self.is_internal_link:
             if '/#narrow/' in self.link:
-                # TODO: Handle internal link.
-                pass
+                self.handle_narrow()
             elif '/user_uploads/' in self.link:
                 # TODO: Open media in the native viewer.
                 pass
         else:
             # TODO: Open external link in the web browser.
             pass
+
+    def handle_narrow(self) -> None:
+        """
+        Handles stream narrow.
+        """
```

Comment:
The commit title (description) for this could be improved - imagine reading just that line after coming back to it - what does this actually do? (without reading the code)